client/model/cluster.go

```diff
@@ -43,45 +43,56 @@ const (
 type ClusterState string
 
 const (
-	// ClusterStatePending is for PENDING state
+	// ClusterStatePending Indicates that a cluster is in the process of being created.
 	ClusterStatePending = "PENDING"
-
-	// ClusterStateRunning is for RUNNING state
+	// ClusterStateRunning Indicates that a cluster has been started and is ready for use.
 	ClusterStateRunning = "RUNNING"
-
-	// ClusterStateRestarting is for RESTARTING state
+	// ClusterStateRestarting Indicates that a cluster is in the process of restarting.
 	ClusterStateRestarting = "RESTARTING"
-
-	// ClusterStateResizing is for RESIZING state
+	// ClusterStateResizing Indicates that a cluster is in the process of adding or removing nodes.
 	ClusterStateResizing = "RESIZING"
-
-	// ClusterStateTerminating is for TERMINATING state
+	// ClusterStateTerminating Indicates that a cluster is in the process of being destroyed.
 	ClusterStateTerminating = "TERMINATING"
-
-	// ClusterStateTerminated is for TERMINATED state
+	// ClusterStateTerminated Indicates that a cluster has been successfully destroyed.
 	ClusterStateTerminated = "TERMINATED"
-
-	// ClusterStateError is for ERROR state
+	// ClusterStateError This state is not used anymore. It was used to indicate a cluster
+	// that failed to be created. Terminating and Terminated are used instead.
 	ClusterStateError = "ERROR"
-
-	// ClusterStateUnknown is for UNKNOWN state
+	// ClusterStateUnknown Indicates that a cluster is in an unknown state. A cluster should never be in this state.
 	ClusterStateUnknown = "UNKNOWN"
 )
 
-// ClusterStateNonRunnable is a list of states in which the cluster cannot go back into running by itself
-// without user intervention
-var ClusterStateNonRunnable = []ClusterState{ClusterStateTerminating, ClusterStateTerminated, ClusterStateError, ClusterStateUnknown}
-
-// ClusterStateNonTerminating is a list of states in which the cluster cannot go back into terminated by itself
-//// without user intervention
-var ClusterStateNonTerminating = []ClusterState{ClusterStatePending, ClusterStateRunning, ClusterStateRestarting, ClusterStateResizing, ClusterStateUnknown}
+var stateMachine = map[ClusterState][]ClusterState{
+	ClusterStatePending:     {ClusterStateRunning, ClusterStateTerminating},
+	ClusterStateRunning:     {ClusterStateResizing, ClusterStateRestarting, ClusterStateTerminating},
+	ClusterStateRestarting:  {ClusterStateRunning, ClusterStateTerminating},
+	ClusterStateResizing:    {ClusterStateRunning, ClusterStateTerminating},
+	ClusterStateTerminating: {ClusterStateTerminated},
+}
```

Comment:
Hey quick question regarding this, would there be a scenario in some sort of lag where when polling the server a cluster that is in a "Pending" and the next state jumps to "Terminated" would that impact any of the logic for clusters?

Reply:
seems like the unit test below says that is the case.

Reply:
it would be a bug of the webapp